WaterJug.py

```diff
@@ -4,101 +4,102 @@
 
 class WaterJug:
 
-    def __init__(self,fileptr):
-        self.num=int(fileptr.readline())
+    def __init__(self,fileptr): #constructor
+        self.num=int(fileptr.readline()) #number of glass
         self.vol=[]
         self.states=[]
-        self.aim=[]
+        self.aim=[] #goal state
         self.connecter=' '
-        volstr=fileptr.readline()[:-1]
-        
+        volstr=fileptr.readline()[:-1] #read start state
+
         for i in volstr:
             if(i!=" "):
-                self.vol.append(int(i))
-                self.states.append(0)
-            
-        self.aim=fileptr.readline()[:-1]
-       
-        self.startState=self.joinStates(self.states,self.connecter)
-        self.mem={self.startState:[]}
-        self.bestPath=[]
+                self.vol.append(int(i)) #assign vol of each glass
+                self.states.append(0) #assign state of 0 to each glass
+
+        self.aim=fileptr.readline()[:-1] #assign goal state
+
+        self.startState=self.joinStates(self.states,self.connecter) # make string with start state "0 0"
+        self.mem={self.startState:[]} #make dictionary with keys "0 0" with value empty array
+        self.bestPath=[] #make empty array for best state
+
         self.compleetSearch()
-        
+
     def findSuccessfullPaths(self,path):
         #print path
-        if path[-1]==self.aim and (len(path)<=len(self.bestPath) or len(self.bestPath)==0):
-            print path;
-            self.bestPath=path[:]
+        if path[-1]==self.aim and (len(path)<=len(self.bestPath) or len(self.bestPath)==0): #if current path edge is the target and current path is shorter than best path or the best path is still empty
+            print path; #print path
```

Comment:
i'll try to make some change to code when i have time.
i see this program that still able to be improved. as you did i also use the program for my college assignment :+1: 
lets hope other people will see it that way, and willing to improve it more :)